kolibri/core/auth/backends.py

```diff
@@ -25,15 +25,23 @@ def authenticate(self, request, username=None, password=None, **kwargs):
         :keyword facility: a Facility object or facility ID
         :return: A FacilityUser instance if successful, or None if authentication failed.
         """
-        users = FacilityUser.objects.filter(username__iexact=username)
         facility = kwargs.get(FACILITY_CREDENTIAL_KEY, None)
+        # First, attempt case-sensitive login
+        user = self.authenticate_case_sensitive(username, password, facility)
+        if user:
+            return user
+
+        # If case-sensitive login fails, attempt case-insensitive login
+        user = self.authenticate_case_insensitive(username, password, facility)
+        return user
+    def authenticate_case_sensitive(self, username, password, facility):
```

Comment:
I think rather than duplicating these methods, we can share logic by passing in `users`, so we would have a method like:

```
def _authenticate_users(self, users, password, facility):
        if facility:
            users = users.filter(facility=facility)
        for user in users:
            if user.check_password(password):
                return user
            # Allow login without password for learners for facilities that allow this.
            # Must specify the facility, to prevent accidental logins
            elif (
                facility
                and user.dataset.learner_can_login_with_no_password
                and not user.roles.count()
                and not user.is_superuser
            ):
                return user
        return None
```

then these two methods:

```
    def authenticate_case_sensitive(self, username, password, facility):
        users = FacilityUser.objects.filter(username=username)
        return self._authenticate_users(users, password, facility)

    def authenticate_case_insensitive(self, username, password, facility):
        users = FacilityUser.objects.filter(username__iexact=username)
        return self._authenticate_users(users, password, facility)
```